### PR TITLE
[Enhancement] (nereids)implement showSqlBlockRuleCommand in nereids

### LINKED ISSUE #42719: [Enhancement] (nereids)implement showSqlBlockRuleCommand in nereids
showSqlBlockRule command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showSqlBlockRule command in nereids:
unsupportedshowSqlBlockRuleStatement
``` SHOW SQL_BLOCK_RULE (FOR ruleName=identifier)?                                ```

### PR DESCRIPTION
**Pull request by Vallishp**
## Proposed changes

Issue Number: close #42719 

implement showSqlBlockRuleCommand in nereids

### CODE REVIEW

**Comment by Vallishp** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowSqlBlockRuleCommand.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        // check auth
+        if (!Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "ADMIN");
+        }
+        ShowSqlBlockRuleStmt showStmt = new ShowSqlBlockRuleStmt(ruleName);
+        List<List<String>> rows = Lists.newArrayList();
+        List<SqlBlockRule> sqlBlockRules = Env.getCurrentEnv().getSqlBlockRuleMgr().getSqlBlockRule(showStmt);
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowSqlBlockRuleCommand.java:66]: pls add a new getSqlBlockRule function without dependence of ShowSqlBlockRuleStmt

Comment (Vallishp) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowSqlBlockRuleCommand.java]: thanks. updated.

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run p0

**Comment by LiBinfeng-01** [comment]
run external

**Comment by Vallishp** [comment]
run build all

**Comment by Vallishp** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/blockrule/SqlBlockRuleMgr.java [hunk 2]
```diff
@@ -90,6 +91,21 @@ public List<SqlBlockRule> getSqlBlockRule(ShowSqlBlockRuleStmt stmt) throws Anal
         return Lists.newArrayList(nameToSqlBlockRuleMap.values());
     }
 
+    /**
+     * Get SqlBlockRule by show command.
+     **/
+    public List<SqlBlockRule> getSqlBlockRule(ShowSqlBlockRuleCommand cmd) throws AnalysisException {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/blockrule/SqlBlockRuleMgr.java:97]: better to use String as parameter and change the old code to call the new function to remove duplicate code
```suggestion
    public List<SqlBlockRule> getSqlBlockRule(String ruleName) throws AnalysisException {
```

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run p0

**Comment by Vallishp** [comment]
run p0

**Comment by Vallishp** [comment]
run p0

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run p0

**Comment by Vallishp** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Comment by Vallishp** [comment]
run buildall

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**